src/App.js

```diff
@@ -53,7 +54,7 @@ const App = () => {
   // onRecaptcha = (token: string) => {
   //   userStorage.setProfileField('recaptcha', token, 'private')
   // }
-  const [useDesktop, setUseDesktop] = useState(store.get('isLoggedIn') === true)
+  const [useDesktop, setUseDesktop] = useState(Config.showSplashDesktop === false || store.get('isLoggedIn') === true)
 
   const continueWithDesktop = () => {
```

Comment:
can be wrapped to useCallback with [setUseDesktop] deps

Reply:
can you explain?

Reply:
`const continueWithDesktop = useCallback(() => setUseDesktop(true), [setUseDesktop]);`

also, as you're memoizing all jsx output in the fact, you could remove useMemo and wrap App function to React.memo

Reply:
But, as actually only {SplashOrRouter} changes i would do so:

```
import React, { memo, useCallback } from 'react'

const DisconnectedSplash = () => (
 <Splash animation={false} />
)

const SplashOrRouter = memo(({ store }) => {
  const isLoggedIn = !!store.get('isLoggedIn')
  const [useDesktop, setUseDesktop] = useState(!Config.showSplashDesktop || isLoggedIn)
  const continueWithDesktop = useCallback(() =>  setUseDesktop(true), [setUseDesktop])

 return (
  <InternetConnection onDisconnect={DisconnectedSplash} isLoggedIn={isLoggedIn}>
    {!isMobile && !useDesktop ? (
      <SplashDesktop onContinue={continueWithDesktop} urlForQR={window.location.href} />
     ) : (
      <RouterSelector />
    )}
    </InternetConnection>
})

const App = () => {
  const store = SimpleStore.useStore()
  useEffect(() => {
    // effect code ...
  }, [])

  return (
  <PaperProvider theme={theme}>
    <SafeAreaView style={styles.safeAreaView}>    
      <SimpleStoreDialog />
      <LoadingIndicator />
      <SplashOrRouter store={store} />
    </SafeAreaView>
  </PaperProvider>
  )
}